File: number3.php
<?php

function count_non_decreasing_series($sequence) {
    $cnt = 0;
    $n = count($sequence);
    $i = 0;
    while ($i < $n - 1) {
        if ($sequence[$i] <= $sequence[$i + 1]) {
            $cnt += 1;
            while ($i < $n - 1 && $sequence[$i] <= $sequence[$i + 1]) {
                $i += 1;
            }
        } else {
            $i += 1;
        }
    }
    return $cnt;
}

function str_to_list($s) {
    return str_split($s);
}

$sequence1 = [5, 1, 2, 3, 2, 5];
echo "Количество неубывающих серий №1: " . count_non_decreasing_series($sequence1) . "\n";

$sequence2 = str_to_list("484756344");
echo "Количество неубывающих серий №2: " . count_non_decreasing_series($sequence2) . "\n";

$sequence3 = str_to_list("12345678998765432123456789987654321");
echo "Количество неубывающих серий №3: " . count_non_decreasing_series($sequence3) . "\n";

?>

File: number3.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        var sequence1 = new List<int> { 5, 1, 2, 3, 2, 5 };
        Console.WriteLine($"Количество неубывающих серий №1: {CountNonDecreasingSeries(sequence1)}");

        var sequence2 = StrToList("484756344");
        Console.WriteLine($"Количество неубывающих серий №2: {CountNonDecreasingSeries(sequence2)}");

        var sequence3 = StrToList("12345678998765432123456789987654321");
        Console.WriteLine($"Количество неубывающих серий №3: {CountNonDecreasingSeries(sequence3)}");
    }

    static int CountNonDecreasingSeries(List<int> sequence)
    {
        int count = 0;
        int n = sequence.Count;
        int i = 0;

        while (i < n - 1)
        {
            if (sequence[i] <= sequence[i + 1])
            {
                count++;
                while (i < n - 1 && sequence[i] <= sequence[i + 1])
                {
                    i++;
                }
            }
            else
            {
                i++;
            }
        }

        return count;
    }

    static List<int> StrToList(string s)
    {
        var result = new List<int>();
        foreach (char c in s)
        {
            result.Add(c - '0');
        }
        return result;
    }
}